### PR TITLE
New hire form flow

### PR DESCRIPTION
**Pull request by RomanTurner**
Flow for businesses to notify RailsDevs when they make a hire
[issue 764](https://github.com/joemasilotti/railsdevs.com/issues/764) 

## Pull request checklist

- [x] My code contains tests covering the code I modified
- [x] I linted and tested the project with `bin/check`
- [x] I added significant changes and product updates to the [changelog](CHANGELOG.md)

### CODE REVIEW

**Comment by RomanTurner** [comment]
I would have like to put the following tests around the nav_bar states:
- [ ] As a business with a developer profile: do both "Found a job?" and "Made a hire?" on the navigation bar show? 
- [ ] As an admin: Does the Admin dropdown navigation menu have the "Hire forms" link?

I don't have a lot of experience with testing, any pointers?

**Comment by RomanTurner** [comment]
Getting this message: 

```bash
✓ Perfect! No inconsistent interpolations found.
✓ Good job! All data is normalized
```
Not sure why it is failing.

**Comment by joemasilotti** [comment]
The best spot for these tests is probably in the navigation bar view component. Play with that for a bit. If you can't figure it out let's pair!

**Comment by joemasilotti** [comment]
Hey @RomanTurner, three week check in. Do you still have appetite to finish this PR?

If not, no worries. I can take it over the finish line.

**Comment by RomanTurner** [comment]
> Hey @RomanTurner, three week check in. Do you still have appetite to finish this PR?
> 
> 
> 
> If not, no worries. I can take it over the finish line.


Definitely, I am just coming back from vacation and can finish the tests up for this. The feature is complete upto that point

**Comment by joemasilotti** [comment]
Excellent! Looking forward to it.

**Comment by RomanTurner** [comment]
@joemasilotti It keeps failing on the i18n-tasks but the console output is: 

```
✓ Perfect! No inconsistent interpolations found.
✓ Good job! All data is normalized
```

**Comment by joemasilotti** [comment]
I think that's the output from normalizing. Can you try running the health check, too?

`bundle exec i18n-tasks health en`

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### March
 
+* March 3 - Added business new hire forms and admin new hire forms
```
Comment (joemasilotti) [inline CHANGELOG.md:7]: Please add the PR # and your handle like the other entries.

File: app/components/nav_bar/user_component.html.erb [hunk 2]
```diff
@@ -11,24 +11,33 @@
                 <%= inline_svg_tag "logo.svg", class: "h-10 w-auto", alt: "RailsDevs logo", title: "RailsDevs logo" %>
               <% end %>
             </div>
-
```
Comment (joemasilotti) [inline app/components/nav_bar/user_component.html.erb:14]: Please add back in these removed lines.

File: app/controllers/hire/forms_controller.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    def form_params
+      params.require(:hire_form).permit(
+        :billing_address,
+        :developer_name,
+        :annual_salary,
+        :position,
+        :start_date,
+        :employment_type,
+        :feedback
+      ).merge(business: current_user.business)
```
Comment (joemasilotti) [inline app/controllers/hire/forms_controller.rb:41]:
```suggestion
      ).merge(business: business)
```

File: app/models/hire/form.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+module Hire
+  class Form < ApplicationRecord
+    include Hire::Notifications
+    self.table_name = "hire_forms"
```
Comment (joemasilotti) [inline app/models/hire/form.rb:4]:
```suggestion
    self.table_name = "hire_forms"
    include Hire::Notifications
```

Code preference :)

File: config/locales/en.yml [hunk 5]
```diff
@@ -523,6 +556,8 @@ en:
       new_developer_notification:
         email_subject: New developer profile added
         title: "%{developer} added their developer profile"
+      new_hire_form_notification:
+        title: "%{business} has hired a developer"
```
Comment (joemasilotti) [inline config/locales/en.yml:560]:
```suggestion
        title: "%{business} hired a developer"
```

File: db/schema.rb [hunk 6]
```diff
@@ -11,8 +11,8 @@
 # It's strongly recommended that you check this file into your version control system.
 
 ActiveRecord::Schema[7.0].define(version: 2023_02_10_162430) do
+
```
Comment (joemasilotti) [inline db/schema.rb:14]: Let's remove all the changes that aren't for the new table and index.

File: test/components/nav_bar/user_component_test.rb [hunk 7]
```diff
@@ -7,11 +7,13 @@ class NavBar::UserComponentTest < ViewComponent::TestCase
...
     render_inline NavBar::UserComponent.new(users(:empty), links: [
       Link.new("Developers", developers_path),
-      Link.new("Pricing", pricing_path)
+      Link.new("Pricing", pricing_path),
+      Link.new("About", about_path)
     ])
 
     assert_link_to developers_path
     assert_link_to pricing_path
+    assert_link_to about_path
```
Comment (joemasilotti) [inline test/components/nav_bar/user_component_test.rb:16]: What's the point of these tests?

File: app/views/admin/hire/forms/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,17 @@
+<%= render Admin::ContainerComponent.new do %>
+  <%= render Admin::ContentComponent.new do %>
+    <%= render Admin::DescriptionListComponent.new(t(".title")) do |list| %>
+      <% list.with_description "Full name" do %>
+        <%= render Admin::BasicLinkComponent.new(@form.business.name, business_path(@form.business)) %>
```
Comment (joemasilotti) [inline app/views/admin/hire/forms/show.html.erb:5]: Let's also add their email address in this same column that links with a `mailto:`.

**Review by joemasilotti — CHANGES_REQUESTED** [review]
This is looking great, nice work Roman!

Outside of the tiny nitpicking code requests, I have to changes:

1. Let's change the two admin buttons to "Developer forms" and "Business forms" because right now "Hire" vs. "Hired" is a little confusing.
2. Let's add a a mailer preview for the new admin email.

**Comment by MikeRayUX** [comment]
Hey guys! Happy to push this one forward!

**Comment by joemasilotti** [comment]
All yours @MikeRayUX - thanks for reaching out on Twitter!

@RomanTurner, thanks for getting most of the work done. I'm planning on awarding both of you the full bounty.

**Comment by MikeRayUX** [comment]
No problem @RomanTurner, thank you for your work on this!

**Pull request closed without merging**